### PR TITLE
update: update readme with more informations

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

/kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area build

> /area driver-kmod

> /area driver-ebpf

> /area libscap

> /area libsinsp

> /area tests

> /area proposals

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,104 @@
 # falcosecurity/libs
 
 As per the [OSS Libraries Contribution Plan](https://github.com/falcosecurity/falco/blob/master/proposals/20210119-libraries-contribution.md), this repository has been chosen to be the new home for **libsinsp**, **libscap**, the **kernel module driver** and the **eBPF driver sources**.
+
```
Comment (leogr) [inline README.md:4]:
```suggestion

These components are at the foundation of [Falco](https://github.com/falcosecurity/falco) and other projects that work with the same kind of data.

This component stack mainly operates on a data source: system calls. This data source is collected using either a kernel module or an eBPF probe, which we call *drivers*. On top of the drivers, libscap manages the data capture process, libsinsp enriches the data, and provides a rich set of API to consume the data. Furthermore, these two libraries also implement a [plugin](https://github.com/falcosecurity/plugins) framework that extends this stack to potentially any other data sources.
```

Tried to improve the introduction a bit. It's just a draft, feel free to change it!

File: README.md [hunk 2]
```diff
@@ -1,3 +1,104 @@
 # falcosecurity/libs
 
 As per the [OSS Libraries Contribution Plan](https://github.com/falcosecurity/falco/blob/master/proposals/20210119-libraries-contribution.md), this repository has been chosen to be the new home for **libsinsp**, **libscap**, the **kernel module driver** and the **eBPF driver sources**.
```
Comment (leogr) [inline README.md:3]:
```suggestion
As per the [OSS Libraries Contribution Plan](https://github.com/falcosecurity/falco/blob/master/proposals/20210119-libraries-contribution.md), this repository has been chosen to be the new home for **libsinsp**, **libscap**, the **kernel module** and the **eBPF probe** sources.
```

File: README.md [hunk 3]
```diff
@@ -1,3 +1,104 @@
...
+reading syscall events from the ring buffer (where they are placed by drivers), and forwarding them up to libsinsp.  
+Moreover, libscap implements OS state collection and supports reading/writing to scap files.  
+* * **libsinsp** (aka: lib for System INSPection) receives events from libscap and enriches them with machine state;  
+moreover, it performs events filtering with rule evaluation through its internal rule engine.  
+Finally, it manages outputs. 
+* * **chisels** are just little lua scripts to analyze an event stream and perform useful actions.  
+In this subfolder, backend code for chisels support can be found.  
+* _proposals/_ folder unexpectedly contains the list of proposals
+* _cmake/modules_ contains the list of cmake modules used during the build for external dep,  
+plus the libscap and libsinsp ones that are used by Falco when building libs.
```
Comment (leogr) [inline README.md:19]:
```suggestion
* [_driver/_](./driver) contains kernel module and eBPF probe source code,
so-called **drivers**.       
* [_userspace/_](./userspace) contains libscap and libsinsp libraries code,
plus chisels related code and common utilities.
  * **libscap** (aka lib for *System CAPture*) is the userspace library
  that directly communicates with the drivers, reading syscall events from
  the ring buffer (where drivers place them), and forwarding them
  up to libsinsp. Moreover, libscap implements OS state collection and
  supports reading/writing to scap files.  
  * **libsinsp** (aka lib for *System INSPection*) receives events from
  libscap and enriches them with machine state: moreover, it performs
  events filtering with rule evaluation through its internal rule engine.
  Finally, it manages outputs. 
  * **chisels** are just little Lua scripts to analyze an event stream
  and perform useful actions. In this subfolder, the backend code for
  chisels support can be found.  
* [_proposals/_](./proposals) unexpectedly contains the list of proposals.
* [_cmake/modules/_](./cmake/modules) contains modules to build
external dependencies, plus the libscap and libsinsp ones; consumers
(like Falco) use those modules to build the libs in their projects.
```

Just formatting and nitpicking :smile_cat:

**Review by leogr — COMMENTED** [review]
This is awesome! :star_struck: 
Thank you @FedeDP :heart: 

I have left some suggestions/improvements, but it already looks good to me!

**Comment by FedeDP** [comment]
Thanks @leogr ! Your suggestions look great too!

File: README.md [hunk 4]
```diff
@@ -1,3 +1,123 @@
...
+Refer to https://falco.org/docs/getting-started/source/ for the list of dependencies.  
+
+Then, simply issue:
+```bash
+cmake ../
+make sinsp
+```
+
+> **NOTE:** using system libraries is useful to cut compile times down, as this way it will only build libs, and not all deps.  
+> On the other hand, system deps version may have an impact, and we cannot guarantee everything goes smoothly while using them.
```
Comment (LucaGuerra) [inline README.md:57]: These parts are not super clear to me. My main question is what kind of output does this generate and in which cases I'll want to do this. While I really like the instructions for the other cases, that is if I `make` the kernel module I'll get a loadable kmod, if I compile the tests I'll be able to run the tests, etc... I don't really know why I would want to compile `sinsp` (+scap) by itself.

**Comment by deepskyblue86** [comment]
It would be nice to have [this](https://falco.org/blog/contribution-drivers-kmod-ebpf-libraries/) referenced somewhere, or at least the image:
<a href="https://falco.org/img/falco-diagram-blog-contribution.png"></a>

**Comment by FedeDP** [comment]
@deepskyblue86 +1 on that! The link is super useful, and the image too!

**Comment by FedeDP** [comment]
Updated! WDYT @deepskyblue86 ?

**Comment by deepskyblue86** [comment]
> Updated! WDYT @deepskyblue86 ?

:+1: :+1: :+1:

**Review by fntlnz — APPROVED** [review]
good job fede!

**Review by leogr — APPROVED** [review]
/approve

:heart_eyes:

**Pull request merged**